### PR TITLE
feat: implement `takeWhile` function

### PR DESCRIPTION
**Pull request by geminicaprograms**
Takes elements from the source as long as the predicate is satisfied. Note that if the predicate fails then subsequent elements are not longer taken even if they could still satisfy it.
Example:
```scala
  Source.empty[Int].takeWhile(_ > 3).toList          // List()
  Source.fromValues(1, 2, 3).takeWhile(_ < 3).toList // List(1, 2)
  Source.fromValues(3, 2, 1).takeWhile(_ < 3).toList // List()
```

### CODE REVIEW

File: core/src/main/scala/ox/channels/SourceOps.scala [hunk 1]
```diff
@@ -141,6 +141,25 @@ trait SourceOps[+T] { this: Source[T] =>
 
   def take(n: Int)(using Ox, StageCapacity): Source[T] = transform(_.take(n))
 
+  /** Takes elements from the source as long as predicate `f` is satisfied (returns `true`). Note that once the predicate `f` is not
```
Comment (adamw) [inline core/src/main/scala/ox/channels/SourceOps.scala:144]: Same as in the others - I would change the wording a bit to indicate that the returned source is a separate one. E.g. `Forwards the elements from this source to the return one, as along ...`

Comment (rucek) [reply core/src/main/scala/ox/channels/SourceOps.scala:144]: We're using "sends to the returned channel" in other operators, so let's use a similar wording here for consistency from the user's perspective.

**Comment by adamw** [comment]
Sorry, needs merge

**Review by adamw — APPROVED** [review]

**Comment by geminicaprograms** [comment]
> Sorry, needs merge

Fixed 👍

**Pull request merged**